docs/docs/service-discovery-load-balancing.md

```diff
@@ -4,51 +4,31 @@ title: Service Discovery & Load Balancing
-ports do not overlap. You can configure this via the `--local_port_min` and `--local_port_max` 
-[command line flags](({{ site.baseurl }}/docs/command-line-flags.html)).
-Marathon does not make use of these assigned service ports itself. 
-It is the responsibility of a separate component to make use of this information and set up load balancing accordingly. 
-A sample script that does this is provided with Marathon, the `haproxy-marathon-bridge`.
-
-Note that service ports should not be confused with the host ports of the individual tasks. 
-Since multiple tasks may run on the same node, host ports are usually assigned dynamically per task
-unless you use `"requirePorts"` or set the `"hostPort"` to non-zero in the docker `"portMappings"`.
-From within the task execution the dynamically assigned host ports are exposed via environment variables. 
-See [task environment variables - host ports]({{ site.baseurl }}/docs/task-environment-vars.html#host-ports) for more information.
-
-Port information for each running task can be queried via the
-[tasks API endpoint]({{ site.baseurl }}/docs/rest-api.html#get-/v2/tasks)
-at `<marathon host>:<port>/v2/tasks`
-
-## Using HAProxy
-Marathon ships with a simple shell script called `haproxy-marathon-bridge`. Alternatively the [marathon-lb](https://github.com/mesosphere/marathon-lb) can be used.
-Both can turn the Marathon's REST API's list of running tasks into a config file for HAProxy, a lightweight TCP/HTTP proxy.
-The `haproxy-marathon-bridge` provides a minimum set of functionality and is easier to understand for beginners.
-Marathon LB on the other hand supports more advanced functionality like SSL offloading, sticky connections and VHost based load balancing.
-
-### haproxy-marathon-bridge
+Once our app is up and running, we need a way to send traffic to it, from other applications on the same cluster and from external clients.
+
+There are several mechanisms available to accomplish this:
+* [Mesos-DNS](https://github.com/mesosphere/mesos-dns) provides service discovery through the domain name system ([DNS](http://en.wikipedia.org/wiki/Domain_Name_System))
+* [Marathon-lb](https://github.com/mesosphere/marathon-lb) provides port based service discovery using HAProxy, a lightweight TCP/HTTP proxy
+* [haproxy-marathon-bridge](https://github.com/mesosphere/marathon/blob/master/examples/haproxy-marathon-bridge) is an example script which configures a local HAProxy installation
+
+For a detailed description of how ports work in Marathon, see [Ports](ports.html).
+
+## Mesos-DNS
+
+Mesos-DNS generates a hostname for each Mesos task (including Marathon application instances) and translates these names to the IP address and port on the machine currently running each application. Mesos-DNS is particularly useful if the apps are launched through multiple frameworks (not just Marathon). Check the Mesos-DNS [documentation and tutorials page](http://mesosphere.github.io/mesos-dns/) for further information.
```

Comment:
It also creates a `SRV` record with which you cannot only lookup the IP but also the port. The latter is necessary if you use random host port assignments. Maybe you could point out that Mesos DNS is especially nice if you use one IP per task and make that routable. (We don't support that well, though)